typescript/src/backoff.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * A convenience matcher for withBackoffRetries that retries irrespective of
+ * the error.
+ *
+ * @param error The error to match against. Not necessarily an Error
+ *        instance, since the retriable function may throw a non-Error.
+ * @returns Always returns true.
+ */
+export function retryAll(error: any): true {
+  return true
+}
+
+/**
+ * A matcher to specify list of error messages that should abort the retry loop
+ * and throw immediately.
+ * @param matchers List of patterns for error matching.
+ * @returns Matcher function that returns false if error matches one of the patterns.
+ *          True is returned if no matches are found and retry loop should continue
+ */
+export function skipRetryWhenMatched(
+  matchers: Array<string | RegExp>
+): ErrorMatcherFn {
+  return (err: unknown): boolean => {
+    if (err instanceof Error) {
+      for (const matcher of matchers) {
+        if (err.message.match(matcher)) {
+          return false
+        }
+      }
+    }
+    return true
+  }
+}
+
+/**
+ * @callback ErrorMatcherFn A function that returns true if the passed error
+ *           matches and false otherwise. Used to determine if a given error is
+ *           eligible for retry in `withBackoffRetries`.
+ * @param {Error} error The error to check for eligibility.
+ * @return {boolean} True if the error matches, false otherwise.
+ */
+export type ErrorMatcherFn = (err: unknown) => boolean
+
+/**
+ * @callback RetrierFn A function that can retry any function passed to it a
+ *           set number of times, returning its result when it succeeds. Created
+ *           using `backoffRetrier`.
+ * @param {function(): Promise<T>} fn The function to be retried.
+ * @return {Promise<T>}
+ */
```

Comment:
Shouldn't we define:
```
export type RetrierFn = ....
```
below this piece of documentation?

Reply:
c26032cd5fe315f00c53727f35494651bdbe5fa9